src/benchmarklib/jcch/jcch_benchmark_item_runner.cpp

```diff
@@ -0,0 +1,324 @@
+#include "jcch_benchmark_item_runner.hpp"
+
+#include <fstream>
+#include <random>
+
+#include "tpch/tpch_queries.hpp"
+#include "utils/string_utils.hpp"
+#include "utils/timer.hpp"
+
+namespace opossum {
+
+JCCHBenchmarkItemRunner::JCCHBenchmarkItemRunner(const std::string& dbgen_path, const std::string& data_path,
+                                                 const std::shared_ptr<BenchmarkConfig>& config,
+                                                 bool use_prepared_statements, float scale_factor)
+    : TPCHBenchmarkItemRunner(config, use_prepared_statements, scale_factor),
+      _dbgen_path(dbgen_path),
+      _data_path(data_path) {
+  _load_params();
+}
+
+JCCHBenchmarkItemRunner::JCCHBenchmarkItemRunner(const std::string& dbgen_path, const std::string& data_path,
+                                                 const std::shared_ptr<BenchmarkConfig>& config,
+                                                 bool use_prepared_statements, float scale_factor,
+                                                 const std::vector<BenchmarkItemID>& items)
+    : TPCHBenchmarkItemRunner(config, use_prepared_statements, scale_factor, items),
+      _dbgen_path(dbgen_path),
+      _data_path(data_path) {
+  _load_params();
+}
+
+std::string JCCHBenchmarkItemRunner::item_name(const BenchmarkItemID item_id) const {
+  Assert(item_id < 22u, "item_id out of range");
+  return std::string("JCC-H ") + (item_id + 1 < 10 ? "0" : "") + std::to_string(item_id + 1);
+}
+
+void JCCHBenchmarkItemRunner::_load_params() {
+  const auto local_queries_path = _data_path + "/queries/";
+  const auto params_path = local_queries_path + "params";
+
+  // Check if the query parameters have already been generated
+  if (!std::filesystem::exists(params_path)) {
+    Timer timer;
+
+    std::cout << "- Creating query parameters by calling external qgen" << std::flush;
+
+    // Check for the existence of dbgen's query templates (1.sql etc.) at the expected location
+    const auto dbgen_queries_path = _dbgen_path + "/queries/";
+    Assert(std::filesystem::exists(dbgen_queries_path),
+           std::string{"Query templates not found at "} + dbgen_queries_path);
+
+    // Create local directory and copy query templates if needed
+    const auto local_queries_dir_created = std::filesystem::create_directory(local_queries_path);
+    if (local_queries_dir_created) {
+      auto cmd = std::stringstream{};
+      cmd << "cd " << local_queries_path << " && ln -s " << _dbgen_path << "/queries/*.sql .";
+      auto ret = system(cmd.str().c_str());
+      Assert(!ret, "Creating symlinks to query templates failed");
+    }
```

Comment:
what happens if create_directory is not successful?

Reply:
Everything goes boom. I added a comment:

```
    std::filesystem::create_directories(jcch_data_path_str);
    // Success of create_directories is guaranteed by the call to fs::canonical, which fails on invalid paths:
    auto jcch_data_path = std::filesystem::canonical(jcch_data_path_str.str() + "bla");
```